message/message-factory.rst

```diff
@@ -1,57 +1,124 @@
 .. _message-factory:
+.. _stream-factory:
 
-Message Factory
-===============
+HTTP Factories
```

Comment:
that was the initial question in #141. i find "Message Factory" wrong here. what should we call it? "Message Factories" is wrong, uri and stream are not messages. other ideas? or should i hunt through the doc and change "message factory" to "http factory" where i find that?